pkg/plugins/policies/meshaccesslog/plugin/xds/configurer.go

```diff
@@ -0,0 +1,147 @@
+	}
+
+	envoyFormat, err = envoyFormat.Interpolate(variables)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to interpolate access log format string with Kuma-specific variables: %s", formatString)
+	}
+
+	return envoyFormat, nil
+}
+
+func (c *Configurer) envoyAccessLog(defaultFormat string) (*envoy_accesslog.AccessLog, error) {
+	var format *api.MeshAccessLog_Format
+	if f := c.Backend.GetFile().GetFormat(); f != nil {
+		format = f
+	} else if f := c.Backend.GetTcp().GetFormat(); f != nil {
+		format = f
+	}
+
+	// TODO json
+
+	if len(format.GetJson()) == 0 {
+		formatString := format.GetPlain()
+		if formatString == "" {
+			formatString = defaultFormat
+		}
+		envoyFormat, err := c.handlePlain(formatString)
+		if err != nil {
+			return nil, err
+		}
+
+		if file := c.Backend.GetFile(); file != nil {
+			return fileAccessLog(envoyFormat.String(), file.Path)
+		} else if tcp := c.Backend.GetTcp(); tcp != nil {
+			path := envoy.AccessLogSocketName(c.Dataplane.Meta.GetName(), c.Mesh)
+			return fileAccessLog(fmt.Sprintf("%s;%s", tcp.Address, envoyFormat.String()), path)
+		}
+	}
+
+	panic("impossible backend type")
+}
+
+func fileAccessLog(format string, path string) (*envoy_accesslog.AccessLog, error) {
+	fileAccessLog := &access_loggers_file.FileAccessLog{
+		AccessLogFormat: &access_loggers_file.FileAccessLog_LogFormat{
+			LogFormat: &envoy_core.SubstitutionFormatString{
+				Format: &envoy_core.SubstitutionFormatString_TextFormatSource{
+					TextFormatSource: &envoy_core.DataSource{
+						Specifier: &envoy_core.DataSource_InlineString{
+							InlineString: format,
+						},
+					},
+				},
+			},
+		},
+		Path: path,
+	}
+
+	marshaled, err := util_proto.MarshalAnyDeterministic(fileAccessLog)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not marshall %T", fileAccessLog)
+	}
+	return &envoy_accesslog.AccessLog{
+		Name: "envoy.access_loggers.file",
+		ConfigType: &envoy_accesslog.AccessLog_TypedConfig{
+			TypedConfig: marshaled,
+		},
+	}, nil
+}
+
+func (c *Configurer) Configure(filterChain *envoy_listener.FilterChain) error {
+	httpAccessLog := func(hcm *envoy_hcm.HttpConnectionManager) error {
+		accessLog, err := c.envoyAccessLog(defaultHttpAccessLogFormat)
+		if err != nil {
+			return err
+		}
+		hcm.AccessLog = append(hcm.AccessLog, accessLog)
+		return nil
+	}
+	tcpAccessLog := func(tcpProxy *envoy_tcp.TcpProxy) error {
+		accessLog, err := c.envoyAccessLog(defaultNetworkAccessLogFormat)
+		if err != nil {
+			return err
+		}
+
+		tcpProxy.AccessLog = append(tcpProxy.AccessLog, accessLog)
+		return nil
+	}
+
+	if err := listeners_v3.UpdateHTTPConnectionManager(filterChain, httpAccessLog); err != nil && !errors.Is(err, &listeners_v3.UnexpectedFilterConfigTypeError{}) {
+		return err
+	}
+	if err := listeners_v3.UpdateTCPProxy(filterChain, tcpAccessLog); err != nil && !errors.Is(err, &listeners_v3.UnexpectedFilterConfigTypeError{}) {
+		return err
+	}
+
+	return nil
+}
```

Comment:
that's a minor nit, but I try to order functions in a way they are executed with public functions at the top, so
```
func Configure()
func envoyAccessLog() // because it's called by Configure
func handlePlain() // because it's called by envoyAccessLog
func fileAccessLog() // because it's called by envoyAccessLog
```

It helps me read the code if I see it the first time.